pkg/kv/kvprober/kvprober.go

```diff
@@ -306,6 +308,26 @@ func (p *Prober) Metrics() Metrics {
 	return p.metrics
 }
 
+// returnLeaseholderInfo searches through the given tracing recording for log messages
+// indicating leaseholder information, extracts leaseholder node ID,
+// and returns this information. Returns an empty string if leaseholder information
+// is not found.
+func (p *Prober) returnLeaseholderInfo(recording tracingpb.Recording) string {
+	// Find the log message containing information about current leaseholder
+	informationLog, isPresent := recording.FindLastLogMessage("node received request")
+
+	if isPresent {
+		// Regular expression to extract leaseholder node ID
+		leaseRegex := regexp.MustCompile(`\[n(\d+)\]`)
+		leaseholder := leaseRegex.FindStringSubmatch(informationLog)
+		// Return leaseholder node ID if found
+		if len(leaseholder) > 0 {
```

Comment:
I suggest asserting that `leaseholder` is the exact size you expect here.

Reply:
Done!